backend/routes.py

```
@@ -1578,11 +1581,23 @@ def return_route(id):
+                logger.error(f"Database error during tool return", exc_info=True, extra={
+                    'operation': 'tool_return',
+                    'tool_id': tool_id,
+                    'user_id': session.get('user_id'),
+                    'error_type': type(e).__name__,
+                    'error_message': str(e)
+                })
                 return jsonify({'error': 'Database error during tool return'}), 500
 
         except Exception as e:
-            print(f"Unexpected error in return route: {str(e)}")
+            logger.error(f"Unexpected error in return route", exc_info=True, extra={
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)